cli_client/python/timesketch_cli_client/commands/search.py

```diff
@@ -0,0 +1,206 @@
+@click.pass_context
+# pylint: disable=too-many-arguments
+def search_group(ctx, query, times, time_ranges, labels, header, output,
+                 return_fields, order, limit, saved_search, describe):
+    """Search and explore."""
+    sketch = ctx.obj.sketch
+    output_format = ctx.obj.output_format
+    search_obj = search.Search(sketch=sketch)
+
+    if output:
+        output_format = output
+
+    new_line = True
+    if output_format == 'csv':
+        new_line = False
+
+    # Construct query from saved search and return early.
+    if saved_search:
+        search_obj.from_saved(saved_search)
+        if describe:
+            describe_query(search_obj)
+            return
+        click.echo(format_output(
+            search_obj, output_format, header), nl=new_line)
+        return
+
+    # Construct the query from flags.
+    search_obj.query_string = query
+    search_obj.return_fields = return_fields
+    search_obj.max_entries = limit
+
+    if order == 'asc':
+        search_obj.order_ascending()
+    elif order == 'desc':
+        search_obj.order_descending()
+
+    # TODO: Add term chips.
+    if time_ranges:
+        for time_range in time_ranges:
+            range_chip = search.DateRangeChip()
+            range_chip.start_time = time_range[0]
+            range_chip.end_time = time_range[1]
+            search_obj.add_chip(range_chip)
+
+    if times:
+        for time in times:
+            range_chip = search.DateRangeChip()
+            range_chip.start_time = time
+            range_chip.end_time = time
+            search_obj.add_chip(range_chip)
+
+    if labels:
+        for label in labels:
+            label_chip = search.LabelChip()
+            if label == 'star':
+                label_chip.use_star_label()
+            elif label == 'comment':
+                label_chip.use_comment_label()
+            else:
+                label_chip.label = label
+            search_obj.add_chip(label_chip)
+
+    if describe:
+        describe_query(search_obj)
+        return
+
+    click.echo(format_output(search_obj, output_format, header), nl=new_line)
+
+
+@click.group('saved-searches')
+def saved_searches_group():
+    """Managed saved searches."""
+
+
+@saved_searches_group.command('list')
+@click.pass_context
+def list_saved_searches(ctx):
+    """List saved searches in the sketch.
+
+    Args:
+        ctx: Click CLI context object.
+    """
+    sketch = ctx.obj.sketch
+    for saved_search in sketch.list_views():
+        click.echo('{} {}'.format(saved_search.id, saved_search.name))
+
+
+@saved_searches_group.command('describe')
+@click.argument('search_id', type=int, required=False)
+@click.pass_context
+def describe_saved_search(ctx, search_id):
+    """Show details for a view.
+
+    Args:
+        ctx: Click CLI context object.
+        search_id: View ID from argument.
+    """
+    sketch = ctx.obj.sketch
+    saved_search = sketch.get_view(view_id=search_id)
```

Comment:
use `sketch.get_saved_search` ... the `get_view` will be deprecated soon enough

also this only checks based on ID, do we want to add the option of a name as well?

Reply:
Done

Added a todo to revisit and add support for saved search name.